### ISSUE TITLE
Cannot select text from deleted lines diff view

### ISSUE THREAD

**Issue by coreh** [microsoft/vscode#8226]
- VSCode Version: 1.3.0-insider
- OS Version: OS X 10.11

Steps to Reproduce:
1. Open a file with changes in inline diff mode
2. Try to select the red (deleted) lines (You can't)

You can select them normally in the side-by-side. Being able to do so is useful to restore/revert small pieces of code manually by copying and pasting.

**Comment by alexandrudima** [comment]
That is a limitation of how the inline diff is implemented. I wonder if we can alleviate this limitation by adding a button somewhere in there (copy or revert this change) that would show up on hover

**Comment by gulshan** [comment]
Instead of buttons, there can be code actions in the diffs for `Previous | Next | Copy original | Revert to original` like the one provided with git merge. I have requested the feature at #30519

**Comment by DrQuailMan** [comment]
With all due respect, please just fix the inline diff implementation.  No need for extra buttons.  If you've had this bug around for a year, you should just make the right fix at this point.

Note that this also prevents ctrl-F from working correctly, since it can't find results in the removed lines.  "Copy original" buttons don't avoid that issue.

**Comment by jenseralmeida** [comment]
This is one of the reasons that I have to open Visual Studio or another diff tool. When traveling the side by side is not an option on small laptops screens and not being able to find or copy the deleted code it is a significant loss.

**Comment by hoangnamitc** [comment]
I think this is a very necessary function !

**Comment by Spown** [comment]
Yes, this limitation basically forces the user to revert the whole batch of changes just to be able to get a single deleted line back. Its not a big deal if the line was just deleted - that is ctrl+z is for, but if I did a bunch of changes since already it's obviously not an option. I often just need to copy one single chunk of text and put it back into the file. Often not even at the location where it was removed from. But I can't.

I think it has something to do with the fact that the removed text is not a part of the surrounding, still present text - so you can't select them together, but I think if the user started selction inside the removed (red) line - he could/should be limited to selecting (and copying) the text only in the same red region. 

![](https://drive.google.com/uc?id=1PhHjneyPiJWs2HMTiOYHTRt0Eo8Tl1KL)

**Comment by bsvipas** [comment]
Any news on this?

/cc @alexandrudima @rebornix @bpasero

**Comment by lukeza** [comment]
A quick workaround is to switch to side by side view that allows copy/paste.

**Comment by cainamaturo** [comment]
This is just painful and makes me so sad.

**Comment by drdebmath** [comment]
It is surprising to see such a common feature request pending from 2016. Or was it solved in some intermediate vscode releases and reappeared again in 1.31? Also right click doesn't work on left side.
Even in side by side view, I was able to select the deleted text, but can't copy it.

**Comment by dufferzafar** [comment]
@drdebmath I searched the VS release notes, I don't think this was ever fixed in earlier versions. There have been multiple such requests over time, so a lot of people do want this.

My current "workaround" is to just have Sublime Text open as well - which I use as my git tool - using its amazing [GitSavvy](https://github.com/divmain/GitSavvy/) plugin.

**Comment by Nyadesune** [comment]
It is a bit frustrating to be able to see deleted text and not being able to select it haha

**Comment by Stanzilla** [comment]
@alexandrudima @rebornix @bpasero any updates here?

**Comment by phifa** [comment]
Can we test/help?

**Comment by johnculviner** [comment]
This was actually the straw that broken the camels back reason for me to keep using WebStorm over VSCode (until WebStorm completely was unable to open a project without crashing ...) This is a SUPER needed feature. I often make changes and want to revert a little bit all the time (via a quick copy paste NOT revert a huge block). Without this ability it's super painful!

**Comment by vitormv** [comment]
Much much needed, please add this.!!

**Comment by Emettant** [comment]
Would have been very useful to have this.

**Comment by titoleiva** [comment]
Please, this is a really important and necessary tool, and maybe a trivial one. Why are you so obstinated?

**Comment by gjsjohnmurray** [comment]
@titoleiva are you able to work around this by using [side-by-side view](https://github.com/microsoft/vscode/issues/8226#issuecomment-399086509)?

**Comment by titoleiva** [comment]
@gjsjohnmurray, yes. It's the best option for now.

**Comment by nbransby** [comment]
Yeah this is really annoying...

**Comment by ShoeQ** [comment]
Not sure how this UX issue has been open since 2016...

**Comment by rebornix** [comment]
You can copy deleted content or deleted line, or revert change by quick actions on the gutter, with latest Stable 1.38.

![actions-for-inline](https://user-images.githubusercontent.com/876920/64294427-a5901900-cf23-11e9-81f6-38862763cd03.gif)

**Comment by powerpaul17** [comment]
But it seems it's still not possible to stage deleted lines as this action does not exist in the gutter menu.

**Comment by jenseralmeida** [comment]
Yes, it is nice to see some improvements, but it took only a couple of usages to see how we still need the full editable capabilities to have a really enjoyable inline diff experience.

Visual studio code is really a pleasure to use, but I really miss my preferred way to work with diffs, fully functional 😉

**Comment by tnrich** [comment]
@rebornix thanks for the update that adds new functionality! It is definitely a step in the right direction! 

That said, it would still be **even better** if we could copy **any portion** of the old diff. 

I find I often only want to copy a small portion of the old diff instead of the entire thing. This is still not possible. 

If you could make it so we could highlight and copy any portion of the old diff that would be ideal!

**Comment by davidhiber** [comment]
Still seems to be an issue in 2019...
Guess its time to go back to webstorm. 🤔

**Comment by geajack** [comment]
Please fix this. I went through a whole file replacing hard-coded strings with references to constants in an internationalization matrix. My intent was to then go back through the diff, copy-pasting the original text in the deleted lines into the matrix. Have to do something a bit more roundabout, now.

**Comment by seanjfellows** [comment]
First comment of 2020! 🤣 😭 

I would make use of this every day if it were fixed...

**Comment by rppudo** [comment]
Still waiting...

**Comment by Kenobi-the-2nd** [comment]
I agree with those here. The ability to select and copy any portion of the text within the inline diff would be very helpful indeed.

**Comment by cyberspectre2** [comment]
Wait, so is it impossible to stage deleted lines in vscode? There's no way to do it?

**Comment by lucasbasquerotto** [comment]
@cyberspectre2 If you want to copy an excerpt of a deleted code and paste in the current one you can do when opening the changes between the 2 files. 

In some cases, this is not so convenient as seeing only the changes of a part of the file when clicking at the left of the page to show the changes of only that location and selecting what you want, even if is only a part of what was deleted (which this issue is addressing, although it already allows to copy the entire part), but for me the full file comparison is still fine in most cases.

**Comment by tedgoddard** [comment]
Please allow us to select and copy any text within the diff.

**Comment by cyberspectre2** [comment]
> @cyberspectre2 If you want to copy an excerpt of a deleted code and paste in the current one you can do when opening the changes between the 2 files.
> 
> In some cases, this is not so convenient as seeing only the changes of a part of the file when clicking at the left of the page to show the changes of only that location and selecting what you want, even if is only a part of what was deleted (which this issue is addressing, although it already allows to copy the entire part), but for me the full file comparison is still fine in most cases.

Not trying to copy deleted text. I want to stage / commit deleted text. Often, a change involves adding text to file 2, but removing text from file 1. In Atom, I can select deleted text from the diff view, and stage that deleted text only. How is this done in vscode?

**Comment by DianaNites** [comment]
Select the lines *around* the deleted ones, right click, stage them

**Comment by marco-m** [comment]
> Select the lines _around_ the deleted ones, right click, stage them

thanks @DianaNites this workaround works and is not that painful!

**Comment by tobiasfeil** [comment]
It's absolutely pathetic and amazes me that this issues has been around for so long and is still not addressed.

I'm actually playing with the idea right now to implement a plugin that uses OCR to extract the text from what is apparently an image of text in the deleted lines and thus makes it highlightable.

This is ridiculous.

**Comment by Artemis21** [comment]
I'm really not sure why this even happens. Surely it is *easier* from a dev point of view to make the text selectable like the rest?
Is it so that when you select both removed and added lines you don't get the github diff problem of them both being merged with no way of telling which is which? That I understand, but still, there should be an easier way of doing it.

**Comment by andrewdant-ert** [comment]
I'd like to throw another voice into the choir saying that this is a highly desirable feature. I'd even go so far as to say it's more valuable to be able to copy deleted lines than new lines, since I have easy access to the new ones on my machine but the deleted ones are only visible to me through git.

**Comment by marcelo-mason** [comment]
The stubbornness of the team is disturbing.  Just paid for this thing too, don't see myself renewing.

**Comment by gjsjohnmurray** [comment]
>  Just paid for this thing too, don't see myself renewing.

@marcelo-mason who did you pay, and how much?

**Comment by tjx666** [comment]
I often delete some code and want to restore it, so this feature is very useful for me.

**Comment by ryancwalsh** [comment]
Like the other 167+ upvotes on this bug from 2016, I'd be super excited for this to be fixed so that I don't have to switch to side-by-side mode to copy the deleted text from the diff. Any plans yet? Thanks!

**Comment by marco-m** [comment]
Hello @kieferrm, I am pinging you because I saw that you are the editor of the current iteration plans and going through this issue I saw that you already scheduled this issue for August 2019. It looks like it slipped through the cracks.
I understand that there are currently 4K issues, but this one is from 2016 and already received a lot of upvotes.

I have two questions for you:
1. (as expected) if possible, please consider re-scheduling this issue :-)
2. more in general, is giving thumbs up to issues a working device to signal community interest? Or are there other methods to follow? I think that this would reduce frustration.

thanks!

**Comment by a7madgamal** [comment]
can someone at least say if opening a PR for this will be accepted? I'm willing to do it if so 🤷

**Comment by Spown** [comment]
> if opening a PR for this will be accepted?

@a7madgamal  [why not](https://github.com/microsoft/vscode/pulls?q=is%3Apr+is%3Aclosed+is%3Amerged)?

**Comment by a7madgamal** [comment]
@Spown well, seeing that this is a 5-year-old issue with a pile of very angry comments I had a feeling they don't want to do this for some reason :D

**Comment by vitormv** [comment]
It's 2021. We've developed multiple mRNA vaccines in record time, shifted whole sectors to home-office, but still cannot easily select portions of an old diff within a web-based code editor. The stubbornness of the team regarding this issue is commendable.

**Comment by Spown** [comment]
@vitormv you know this is a free product, right? why is it always self entitled spoiled people using current-year-argument?

**Comment by maksnester** [comment]
Can someone recommend an extension that fixes it? This is the vscode way of fixing everything after all... just download a new extension.

**Comment by shenguanjiejie** [comment]
The most sad issue for VSCode.

**Comment by Avivhdr** [comment]
![line diff](https://user-images.githubusercontent.com/23265867/146250536-ed9c37e4-1d57-41d6-b33d-b06efed06a5d.gif)

**Comment by kamerat** [comment]
Although that works @Avivhdr, it's not really user friendly. Many times I just want to copy parts of the changes. It is then much simpler to just select/highlight what you want and copying it that "normal" way. I don't think you deserve down votes for providing a workaround though. This might very well work for someone. However, I do not think the PR should be closed because of it.

**Comment by Avivhdr** [comment]
I agree 💯 %.
Of course it's not ideal. 
I'm eying this issue for a while now and waiting for it to be resolved.
Just wanted to let people know, in case they don't, that this way exists and can be used, until a PR that solves it will be merged.

**Comment by tnrich** [comment]
The ability to copy the deleted line was already mentioned on this thread here: https://github.com/microsoft/vscode/issues/8226#issuecomment-528103843

**Comment by mycahmason** [comment]
Just highlight the lines above and below the deleted line(s) ...since you obviously can't select lines that have been deleted since they no longer exist. It's not intuitive but it works great, I just did it after having the same problem.

**Comment by ptim** [comment]
> A quick workaround is to switch to side by side view that allows copy/paste. 
> [#issuecomment-399086509](https://github.com/microsoft/vscode/issues/8226#issuecomment-399086509)

Sweet workaround, thanks!

I've added a keybinding that toggles inline view only when in a diff:

```json
{
    "key": "cmd+i",
    "command": "toggle.diff.renderSideBySide",
    "when": "editorTextFocus && isInDiffEditor"
  },
```

**Comment by tinder-tannerbennett** [comment]
How has this issue been open since 2016?!

**Comment by motcke** [comment]
Any plans to fix this?

**Comment by jd-solanki** [comment]
I just started using a vertical monitor and I can't use side-by-side diff so I changed it to inline and boom. I can't select the text anymore 😭

**Comment by sickdyd** [comment]
Was a bit hasty, reverted my change to be able to copy the code that I wanted to use somewhere else. Forgot to re-revert it and ended up banging my head against a wall for quite a while before realizing what happened. Please fix it!

**Comment by guettli** [comment]
Hello vscode team, this issue has 278 thumb-up votes. Is there anything that I can do? I am willing to donate 50$, if some other users do the same, we should have enough money to pay a developer to fix this.

**Comment by viridium** [comment]
guettli: the more the ticket is open, the more its value goes up…

**Comment by viridium** [comment]
Maybe the deleted text is not selectable because it's, well, deleted? *badabing*

**Comment by viridium** [comment]
"This ticket has been in the queue for too long and will be automatically closed."

**Comment by MichelleRos** [comment]
This really is a very frustrating issue... Would be nice if someone on the team could just spend the 2 minutes to chime in here and explain why they haven't fixed an issue that so many users have had problems with - is there a technical limitation that makes it not possible to fix? @alexdima @rebornix @bpasero 

The option to right-click and copy the changed line is useful, however this does not solve the problem, and seems like a workaround of some kind.

**Comment by jojorne** [comment]
Well, I want it too and I gave my reason here: https://github.com/microsoft/vscode/issues/177577

**Comment by tnrich** [comment]
Whoever fixes this is going to be a straight up hero lol 😅🙏💥

**Comment by viridium** [comment]
Can someone share the general approach to doing it? We may be able to pick it up. Could it be as simple as marking a DOM element selectable in JavaScript or even HTML?

**Comment by ix-anemoff** [comment]
This bug is open since 2016. This can't be that long to fix.

**Comment by tjx666** [comment]
@hediet Will this be fixed in diff editor v2?

**Comment by Immorpher** [comment]
I hope this does get resolved eventually, it would save me so much time.

**Comment by kozzztya** [comment]
I've installed GitLens, there you can Toggle File Changes and on hover you can copy prev changes. 

<img width="481" alt="Screenshot 2023-07-20 at 14 30 23" src="https://github.com/microsoft/vscode/assets/7671659/8f324b7b-f687-4cc0-91f3-b6b9806534bc">

If it's possible to do in the extension, I'm sure it's possible to implement in vscode.

**Comment by DiogoAluai** [comment]
Let's go for the 10-year mark! :D

**Comment by hongwen000** [comment]
LLLLLove to GitLens!!!!

**Comment by SBokutsWA** [comment]
hi! any updates on this one? is really annoying to click `icon -> copy changed the line `

in many cases, you need only part of it and it should be able to just select text and copy it

**Comment by sergei-dyshel** [comment]
@SBokutsWA
Not to diminish the importance of this issue (it drives me crazy too), but you can just temporarily change to side-by-side diff for copying. I have a keyboard shortcut assigned to this.  And with the recent "use inline view when space is limited" option you don't even have to toggle - just resize the window.

**Comment by SBokutsWA** [comment]
@sergei-dyshel 

yes, thanks for the info. I found it too. looks like a temporary solution that works. 

But anyway it would be nice to have the same behavior inside incline view

**Comment by goldy1992** [comment]
As a workaround for git users:

Just run `git diff $CHANGED_FILE_PATH` from terminal and just copy the lines from your git editor, where  `$CHANGED_FILE_PATH` is the path to the file you need the deleted lines from.

**Comment by Spown** [comment]
> just

@goldy1992  🤡

**Comment by shinspiegel** [comment]
I'm facing the same issue, I was going to create a new issue for this topic. 
I do recall that I was able to select the deleted text in the diff, but I can't. 

How can I revert this?
How can I make this feature enabled again?

**Comment by Spown** [comment]
> I do recall that I was able to select the deleted text in the diff, but I can't.

@shinspiegel switch to the side by side, i.e. turn off the inline mode
![image](https://github.com/microsoft/vscode/assets/3817672/8fbe1ee5-94be-4aa7-9046-91723f548896)

**Comment by shinspiegel** [comment]
> > I do recall that I was able to select the deleted text in the diff, but I can't.
> 
> @shinspiegel switch to the side by side, i.e. turn off the inline mode 

Holy mother, I was weeks struggling with this editor to make this simple task.
From the bottom of my heart, thanks @Spown

**Comment by gabefighter99** [comment]
The best workaround I was able to come up with was right-click -> Copy changed lines. But this is not a proper solution. If the changes are visible I don't understand why they can't also be selectable.

**Comment by andrew-hossack** [comment]
Still waiting on this, please.

**Comment by shinspiegel** [comment]
> Still waiting on this, please.

I waited for 2 years and got no good answer, so I moved my stuff to NeoVim.
I would recommend you to do a similar move, maybe this would shake Microsoft to take some action instead of pushing more AI on the editor.

**Comment by avonar** [comment]
i waited for 7 =)

**Comment by kapple19** [comment]
I don't understand why the developers would take the time to implement things like a voice dictator in the editor, but won't fix issues like this that have been requested for a very long time. Is it high time to move to another editor?

**Comment by russeg** [comment]
I also do not understand why this has not been addressed, or at least given an explanation on why. 

this is really a great mystery.

**Comment by shinspiegel** [comment]
@russeg 
Because of this, and others "not aligned with the community" ideas from this text editor (or IDE, I don't know anymore), 
I changed to another text editor. Sad, but for me I already said goodbye,

**Comment by NSExceptional** [comment]
@hediet if you can point me to the file that handles this, I can try to fix it myself and make a PR.

**Comment by dortamiguel** [comment]
First time I see a text editor that can select text that can't copy, super annoying

**Comment by sergei-dyshel** [comment]
While I'm fully 👍-ing this request and personally would love to see this feature implemented, I'm subscribed to this thread for a long  and some comments people leave here are pretty hilarious (no offence intended), like "prevents me from using vscode", "refactoring is impossible", "moved to another editor because of that". Is it so hard to temporarily switch to side-by-side view and copy the needed text? You can even add a keyboard shortcut for that, to speed-up the flow.

With that said, I see this as only a small part of a much bigger problem which is much more limited implementation of deleted chunks in inline editor as compared to side-by-side implementation. One much worse manifestation of it is https://github.com/microsoft/vscode/issues/164729, which is really a bug and not just missing functionality. It is closed as *fixed* but actually it's not. 

Also looking at [this query](https://github.com/microsoft/vscode/issues?q=is%3Aissue+is%3Aopen+label%3Afeature-request+sort%3Areactions-%2B1-desc+label%3A%22papercut+%3Adrop_of_blood%3A%22), this seems to be most +1-ed feature requests taged with "papercut" tag so I assume that the fix is difficult otherwise it would be done long ago. 

So I'm surprised that one from vscode team gave a proper explanation of the problem and team's thoughts on fixing it. 
@hediet, @alexr00 may be you can invest some time and describe the challenge?

**Comment by hediet** [comment]
> may be you can invest some time and describe the challenge?

The challenge is that we would have to create a new code editor for every deleted block. This scales very poorly and might cause freezes when a diff contains many (probably >20) deleted/modified blocks. I actually have a prototype, but without virtualization this is not going to work.

**Comment by sergei-dyshel** [comment]
@hediet 

Curious, what kind of virtualization do you mean here?
I understand that in side-by-side diff both sides are implemented as full-blown text editor widget. So what happens in inline mode? Can't we have something like "sparse" editor that renders subset of line (the deleted ones)? Can you point to code that implements side-by-side and inline diff views?

**Comment by hediet** [comment]
> Can't we have something like "sparse" editor that renders subset of line

That is quite involved!

> So what happens in inline mode?

We use the line-rendering logic of the editor to render the deleted lines, but we don't create an widget for it. Thus there is no selection/copy/etc. support.

**Comment by samfundev** [comment]
Apologizes in advance, I do not know if my solution has already been considered and I'm going based on some assumptions of what I am guessing is happening internally.

If I understand it correctly, currently a diff editor is based on some source file that is being edited. So for added lines it's as simple as highlighting them as they already exist in the source file. But it's not the same for deleted lines because they don't exist in the source file. So they have to rendered in manually. I assume this is how you are able to edit a diff, because you're just editing the source file.

My solution would be to compute the text content necessary to display a diff and put that into an editor instead. This way added or deleted lines could be highlighted the same way. Mark the deleted lines read only. Whenever the user makes an edit, apply it to the source file instead.

**Comment by hediet** [comment]
> My solution would be to compute the text content necessary to display a diff and put that into an editor instead. This way added or deleted lines could be highlighted the same way. Mark the deleted lines read only. Whenever the user makes an edit, apply it to the source file instead.

Interesting idea, but then side-by-side and inline view would have to operate on different text models.
Also, you would need to project edits in this virtual text model back to the original one. That is very tricky.

**Comment by starball5** [comment]
Related on Stack Overflow: [Enable text selection on changed lines on VS Code Open Changes editor tab](https://stackoverflow.com/q/78413438/11107541)

**Comment by freewind** [comment]
8 years

**Comment by Nagell** [comment]
Yay, it's growing so fast ;)

**Comment by matthias-ccri** [comment]
_**LET ME SELECT THE TEXT, VSCODE!!!!!!!**_

**Comment by MDerman** [comment]
Literally please add this feature.

Another reason to appreciate Rider 🤷‍♂️

**Comment by barrytdh** [comment]
2025, I’m still here waiting for VSCode to let me select the text. 🙃

**Comment by krutoo** [comment]
The answer on StackOverflow is helped me to return deleted lines selection:
https://stackoverflow.com/questions/78413438/enable-text-selection-on-changed-lines-on-vs-code-open-changes-editor-tab

But to enable Side-by-side view you should disable "Use inline view when space is limited"

<img width="436" alt="Image" src="https://github.com/user-attachments/assets/072c4cd6-ff56-4e25-8c69-754b02b40ad6" />

**Comment by matthias-ccri** [comment]
To be clear, the above comment should _**not**_ be considered an acceptable solution to this issue. Sure, we can figure out all sorts of roundabout ways to copy the text. But what this issue is asking for is to just allow the text to be selected and copied.

**Comment by yalopov** [comment]
Can't believe this hasn't been implemented yet, so I started using IntelliJ IDEA. Unsubscribing from this as probably won't be even considered. Cheers.

**Comment by tnrich** [comment]
Added a 100$ bounty for this issue to gitpay - https://gitpay.me/#/task/1064

 🙏 🙏 Can't wait to pay that out to the person/people who solve this one! 🙏 🙏

**Comment by rwspeight** [comment]
I'm absolutely flabbergasted this is even a thing and it's been known for so long.  I was just doing some refactoring and realized I needed a small block of code within some deletions.  Of course I thought to myself "The diff is one click away.  I'll grab it from there!"  I clicked the margin change indicator and quickly found I could only look at the code block I wanted.  The frustration is compounded by the fact I _can_ select the green text but not the red.  It's absurd for an app, whose most core feature is the display and editing of text, to not allow one of the most essential interactions with that text.

**Comment by amit-s19** [comment]
@tnrich I've raised a PR regarding this #243399, please have a look

**Comment by tnrich** [comment]
@amit-s19 if that change makes it in and indeed fixes this issue I will be more than happy to award you the bounty! Gotta say if the fix is just a one liner I'll be pretty blown away though 😅🙏

**Comment by antoninkriz** [comment]
Added another $20 to the bounty from @tnrich. This issue is driving me nuts for years. Again and again.

**Comment by DenianFossatti** [comment]
@tnrich @amit-s19 I created a PR to fix this issue #248058, hope it will solve the issue

**Comment by jayantpranjal0** [comment]
@coreh is this issue open to work upon??

**Comment by david-meri** [comment]
Hi @coreh, @tnrich, @hediet 

Implement text selection in deleted lines for inline diff view:

Implementing text selection functionality for deleted lines in the inline diff view would be necessary. This requires modifying the code editor's diff view logic:

  * Identify the responsible code section that handles text selection in the inline diff view.
  * Update the code to allow the selection of deleted lines (red lines) by modifying the selection logic to recognize deleted line elements as selectable.
  * Ensure that the new selection behavior doesn't interfere with other inline diff view functionality, such as editing non-deleted text or navigating the view.

After implementing this change, users should be able to select text from deleted lines in the inline diff view, enabling them to restore or revert code pieces more easily.

Thanks

**Comment by Keavon** [comment]
More AI generated meaningless text to reduce the signal-to-noise ratio on the internet...

**Comment by ricky-orchid** [comment]
In the meantime can we please add another option to the menu popup when clicking on the deleted line:  

"Blame - Bruce Lee (1 week ago)"  - this should open up the blame commit diff for the deleted line

**Comment by nonnapopoa** [comment]
Finally we will have a useful feature. I was getting tired of seeing AI features in every VS Code release.
